web-common/src/features/entity-management/resource-invalidations.ts

```diff
@@ -103,13 +103,24 @@ async function invalidateResource(
 
   switch (resource.meta.name.kind) {
     case ResourceKind.Source:
-    // eslint-disable-next-line no-fallthrough
+      if (resource.source?.state?.table)
+        // make sure table is populated
+        return invalidateProfilingQueries(
+          queryClient,
+          resource.meta.name.name,
```

Comment:
I'm seeing some type issues in my editor: 
![image](https://github.com/rilldata/rill/assets/14206386/9fe380e1-09f0-429e-992b-1272fb1b59d5)

Reply:
There are a bunch of errors like this across the app. These are because a setting was enabled to add strict mode and orval adds `?` to all fields even the ones that are not optional in the proto definition. This is one of those that can be ignored.